### PR TITLE
feat(ci): setup automated rebuild on base image bumps

### PR DESCRIPTION
**Pull request by hutchic**
A rehash on https://github.com/Kong/docker-fpm/pull/10 this time not using Docker Hub automated builds.

- It added a second build system without a lot of advantages
- It meant building to test vs building to release were different processes

**Desired Result**
base image updates automatically rebuild our image

**Prerequisites**

- [x] DOCKERHUB_TOKEN secret setup in github
- [x] DOCKERHUB_USERNAME secret setup in github
- [x] AUTO_MERGE_TOKEN secret as a **dependabot** secret
- [x] in github branch protection rules require approval
- [ ] in github branch protection rules require build checks to pass
- [x] in github options enable auto-merge

**Approach**
- explicitely pin our base image desired sha's
- setup dependabot such that it makes PR's to update the SHA's
- setup automerge for dependabot PR's
- setup autorelease

**Testing**

I ran this CI on my fork of the repository in my docker hub account. The result can be viewed https://github.com/hutchic/docker-fpm/pull/18 which demonstrates dependabot opened a PR, did a test build that passed, approved / merged the PR, when it landed on `main` did a test build that passed, generated a release (github tag, github release, comment and label PR's in the release) and finally pushed relevant images to docker hub

image

image

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -33,22 +22,33 @@ jobs:
...
+      uses: ahmadnassri/action-semantic-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     - name: Docker meta
+      if: steps['release']['outputs']['published'] == 'true'
       id: meta
       uses: crazy-max/ghaction-docker-meta@v3
       with:
-        images: kong/fpm
+        images: hutchic/fpm
```
Comment (hutchic) [inline .github/workflows/release.yml:41]: swap back to `kong/`

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,19 @@
+FROM amd64/ubuntu:xenial@sha256:a3785f78ab8547ae2710c89e627783cfa7ee7824d3468cae6835c9f4eae23ff7
+RUN echo "disabled"
+
+FROM arm64v8/ubuntu:xenial@sha256:70fa660340a344b46cc56b3606dc8abd3bf48b5cbce13d01c720e9793a6bc3c0
+RUN echo "disabled"
+
+FROM arm32v7/ubuntu:xenial@sha256:b722e2654241f9681f4719dce7aa16a2f0c35769e17a636f5b39a33967d1aeb8
```
Comment (curiositycasualty) [inline Dockerfile:7]: @hutchic can you write to me in finger words here (again) why we're stuck at xenial (if intentional)?

**Comment by curiositycasualty** [comment]
Barring `xenial`, this seems good to :shipit:

File: Dockerfile (hunk 2 above)
Comment (hutchic) [reply Dockerfile:7]: we _might_ not be explicitely stuck at xenial it just needs to be a deb based docker in general to cover rpm / deb options. For reasons it's kind of challenging to use rpm distributions to make deb but not vice versa

Comment (curiositycasualty) [reply Dockerfile:7]: Hmm okay. I guess we can address an upgrade beyond `xenial` here in later work.

**Review by curiositycasualty — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (hutchic) [inline .github/workflows/release.yml:41]:
```suggestion
        images: kong/fpm
```

**Pull request merged**